capstone/src/main/webapp/script.js

```diff
@@ -1,31 +1,153 @@
 function createMap() {
-    const map = new google.maps.Map(
-        document.getElementById('map-container'),
-        {center: {lat: 37.7749, lng: -122.4194}, zoom: 12}
-    );
-  
-    //**adding zipcode overlay */
-    map.data.loadGeoJson('zipcode-data.json');
-    //**adding precinct overlay */
-    map.data.loadGeoJson('neighborhoods.json');
-    
-    map.data.setStyle({visible: false});
-
-    const cityBorder = [
-        {lat: 37.708305, lng: -122.502691},
-        {lat: 37.708229, lng: -122.393322}
-    ]
-
-    const cityLimit = new google.maps.Polygon({
-        paths: cityBorder,
-        strokeColor: 'black',
-        strokeOpacity: 0.8,
-        strokeWeight: 2,
-        fillColor: 'black',
-        fillOpacity: 0.35
-    });
-  
+  const map = new google.maps.Map(document.getElementById('map-container'),
+      {
+        center: {lat: 37.7749, lng: -122.4194},
+        zoom: 12,
+      });
+
+  const cityBorder = [
+    {lat: 37.708305, lng: -122.502691},
+    {lat: 37.708229, lng: -122.393322},
+  ];
+
+  const cityLimit = new google.maps.Polygon({
+    paths: cityBorder,
+    strokeColor: 'black',
+    strokeOpacity: 0.8,
+    strokeWeight: 2,
+    fillColor: 'black',
+    fillOpacity: 0.35,
+  });
+
   cityLimit.setMap(map);
+
+  //* *button for centering map*/
+  const centerControlDiv = document.createElement('div');
+  const centerControl = new CenterControl(centerControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(centerControlDiv);
+
+  //* *button for zipcode data layer */
+  const zipControlDiv = document.createElement('div');
+  const zipControl = new ZipControl(zipControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(zipControlDiv);
+
+  //* *button for zipcode data layer */
+  const precinctControlDiv = document.createElement('div');
+  const precinctControl = new PrecinctControl(precinctControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER]
+      .push(precinctControlDiv);
+}
+
+function CenterControl(controlDiv, map) {
+  //* *button creation and positioning*/
+  const controlUI = document.createElement('div');
+  controlUI.style.backgroundColor = '#fff';
+  controlUI.style.border = '2px solid #fff';
+  controlUI.style.borderRadius = '3px';
+  controlUI.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+  controlUI.style.cursor = 'pointer';
+  controlUI.style.marginBottom = '22px';
+  controlUI.style.textAlign = 'center';
+  controlUI.title = 'Click to recenter the map';
+  controlDiv.appendChild(controlUI);
+
+  //* *css for interior of all buttons*/
+  const text = document.createElement('div');
+  text.style.color = 'rgb(25,25,25)';
+  text.style.fontFamily = 'Roboto,Arial,sans-serif';
```

Comment:
I agree with Prashanti that it would be nice if the css was in the stylesheet rather than the script file, but given the time crunch that we are in, this should suffice

Reply:
I could delete the style code and have it look how it automatically does, if that would be better for you guys.

Reply:
Yea FWIW, if the styling is static, you would probably want it in its own CSS file. Following this practice makes it easier for others to grok where the styling is set as a codebase grows in complexity and keeps a clear separation of concerns.